src/Goetas/Twital/Attribute/AttrAttribute.php

```diff
@@ -13,9 +14,9 @@
  */
 class AttrAttribute implements Attribute
 {
-    public static function getVarname(\DOMNode $node)
+    public static function getVarname(\DOMElement $node)
     {
-        return "__a9" . strtr(md5($node->ownerDocument->saveXML($node)).spl_object_hash($node), "-","_");
+        return "__a9" . strtr($node->getAttributeNS(Twital::NS, '__internal-id__').spl_object_hash($node), "-","_");
```

Comment:
It is also possible to create plugins that are able to "clone" dom nodes. Cloning a node, the `__internal-id__` will remain untouched, so using both `spl_object_hash` and `__internal-id__` the node identifier "should" be more reliable.